homeassistant/components/media_player/emby.py

```diff
@@ -4,252 +4,227 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/media_player.emby/
 """
+import asyncio
 import logging
 
-from datetime import timedelta
-
 import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
 from homeassistant.components.media_player import (
-    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
-    SUPPORT_SEEK, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK, MediaPlayerDevice,
-    SUPPORT_PLAY, PLATFORM_SCHEMA)
+    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE, SUPPORT_SEEK, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK,
+    MediaPlayerDevice, SUPPORT_PLAY, PLATFORM_SCHEMA)
 from homeassistant.const import (
-    CONF_HOST, CONF_API_KEY, CONF_PORT, CONF_SSL, DEVICE_DEFAULT_NAME,
-    STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING, STATE_UNKNOWN)
-from homeassistant.helpers.event import (track_utc_time_change)
-from homeassistant.util import Throttle
+    STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING,
+    CONF_HOST, CONF_PORT, CONF_SSL, CONF_API_KEY, DEVICE_DEFAULT_NAME,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
 import homeassistant.util.dt as dt_util
 
-REQUIREMENTS = ['pyemby==0.2']
+REQUIREMENTS = ['pyemby==1.0']
+
+_LOGGER = logging.getLogger(__name__)
 
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
-MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+CONF_AUTO_HIDE = 'auto_hide'
 
 MEDIA_TYPE_TRAILER = 'trailer'
+MEDIA_TYPE_GENERIC_VIDEO = 'video'
 
+DEFAULT_HOST = 'localhost'
 DEFAULT_PORT = 8096
+DEFAULT_SSL = False
```

Comment:
Okay, let's get some things straight here. (and we should probably add this to some documentation somewhere)

@pvizeli is right that less security is indeed opt-out and not opt-in. @mezz64 is also right however, we also don't want to be broken by default.

There are a few different SSL options in use in Home Assistant. Let me clarify them so we're all on the same page:

 - `VERIFY_SSL`: option that is used to tell platform to connect over SSL but not verify the cert (ugh). This should never be `False` by default.
 - `PROXY_SSL`: option to tell platform that user is hosting it's own proxy in front of the service and thus we need to connect over SSL. Should be `False` by default because it is a user customization about how the service is exposed.
 - `USE_SSL`: option to tell platform to connect over SSL. This, in my opinion, should match the default setting of the service.

The last one is the one we're talking about here. Since Emby will not have a valid certificate out of the box when turning on SSL, we should default SSL to `False`.